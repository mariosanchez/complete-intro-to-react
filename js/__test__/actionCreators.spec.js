// @flow
import moxios from 'moxios';
import { setSearchTerm, addAPIData } from '../actionCreators';
import getAPIData from '../asyncActions';

const show = {
  title: 'Silicon Valley',
  year: '2014â€“',
  description:
    'Follows the struggle of Richard Hendricks, a silicon valley engineer trying to build his own company called Pied Piper.',
  poster: 'sv.jpg',
  imdbID: 'tt2575988',
  trailer: '69V__a49xtw',
};

test('setSearchTerm', () => {
  expect(setSearchTerm('New York')).toMatchSnapshot();
  // This is the same as we do this
  // expect(setSearchTerm('New York')).toEqual({type: 'SET_SEARCH_TERM', payload: 'New York'});
  // Beacaus snapshots are json structures as well
});

test('addAPIData', () => {
  expect(addAPIData(show)).toMatchSnapshot();
});

test('getAPIData', (done: Function) => {
  const dispatchMock = jest.fn();
  moxios.withMock(() => {
    getAPIData(show.imdbID)(dispatchMock);
  });
  moxios.wait(() => {
    const request = moxios.requests.mostRecent();
    request
      .respondWith({
        status: 200,
        response: show,
      })
      .then(() => {
        expect(request.url).toEqual(`http://localhost:3000/${show.imdbID}`);
        expect(dispatchMock).toBeCalledWith(addAPIData(show));
        done();
      });
  });
});
