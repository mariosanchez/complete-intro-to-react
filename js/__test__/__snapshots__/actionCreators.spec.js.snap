// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addAPIData 1`] = `
Object {
  "payload": Object {
    "description": "Follows the struggle of Richard Hendricks, a silicon valley engineer trying to build his own company called Pied Piper.",
    "imdbID": "tt2575988",
    "poster": "sv.jpg",
    "title": "Silicon Valley",
    "trailer": "69V__a49xtw",
    "year": "2014â€“",
  },
  "type": "ADD_API_DATA",
}
`;

exports[`setSearchTerm 1`] = `
Object {
  "payload": "New York",
  "type": "SET_SEARH_TERM",
}
`;
